src/transformers/models/grounding_dino/modeling_grounding_dino.py

```diff
@@ -1914,7 +1914,9 @@ def forward(
                 reference_points_input = reference_points[:, :, None] * valid_ratios[:, None]
             else:
                 raise ValueError("Last dim of reference_points must be 2 or 4, but got {reference_points.shape[-1]}")
-            query_pos = get_sine_pos_embed(reference_points_input[:, :, 0, :], num_pos_feats=self.config.d_model // 2)
+            query_pos = get_sine_pos_embed(
+                reference_points_input[:, :, 0, :], num_pos_feats=self.config.d_model // 2
+            ).to(vision_encoder_hidden_states.dtype)
```

Comment:
Same here - if `reference_points_input` was created in the correct dtype upstream then we wouldn't need to recast here